blocks/api/raw-handling/utils.js

```diff
@@ -152,6 +152,15 @@ export function isInvalidInline( element ) {
 		return false;
 	}
 
+	// an image inside an anchor is not invalid inline
```

Comment:
Suggestion here https://github.com/WordPress/gutenberg/pull/5487/files#r173170195 would also avoid this check.